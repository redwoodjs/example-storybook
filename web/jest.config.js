// More info at https://redwoodjs.com/docs/project-configuration-dev-test-build

const config = {
  rootDir: '../',
  preset: '@redwoodjs/testing/config/jest/web',
  verbose: true,
  moduleNameMapper: {
    // mock redwood magic webpack aliases
    // see: https://github.com/redwoodjs/redwood/blob/main/packages/testing/config/storybook/preview.js
    '~__REDWOOD__USER_WEB_DEFAULT_CSS': 'identity-obj-proxy',
    '~__REDWOOD__USER_STORYBOOK_PREVIEW_CONFIG':
      '<rootDir>/web/config/storybook.preview.js',
  },
  transform: {
    '^.+\\.mdx?$': '@storybook/addon-docs/jest-transform-mdx', // https://github.com/storybookjs/storybook/issues/11376#issuecomment-757417250
  },
  setupFilesAfterEnv: [`<rootDir>/web/test/mocks.js`],
}

module.exports = config
