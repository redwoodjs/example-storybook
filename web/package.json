{
  "name": "web",
  "version": "0.0.0",
  "private": true,
  "browserslist": {
    "development": [
      "last 1 version"
    ],
    "production": [
      "defaults",
      "not IE 11",
      "not IE_Mob 11"
    ]
  },
  "dependencies": {
    "@redwoodjs/forms": "0.49.0",
    "@redwoodjs/router": "0.49.0",
    "@redwoodjs/web": "0.49.0",
    "prop-types": "15.8.1",
    "react": "17.0.2",
    "react-dom": "17.0.2"
  }
}
