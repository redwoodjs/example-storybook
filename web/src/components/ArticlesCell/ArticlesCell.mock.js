// Define your own mock data here:
export const standard = () => ({
  articles: [
    {
      id: 42,
      title: 'ArticlesCell Title 1',
      body: 'ArticlesCell Body 1',
      createdAt: '2020-01-01T00:00:00.000Z',
    },
    {
      id: 43,
      title: 'ArticlesCell Title 2',
      body: 'ArticlesCell Body 2',
      createdAt: '2020-01-02T00:00:00.000Z',
    },
    {
      id: 44,
      title: 'ArticlesCell Title 3',
      body: 'ArticlesCell Body 3',
      createdAt: '2020-01-03T00:00:00.000Z',
    },
  ],
})
