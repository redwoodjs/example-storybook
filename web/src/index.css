/**
 * START --- SETUP TAILWINDCSS EDIT
 *
 * `yarn rw setup ui tailwindcss` placed these imports here
 * to inject Tailwind's styles into your CSS.
 * For more information, see: https://tailwindcss.com/docs/installation#include-tailwind-in-your-css
 */
@import 'tailwindcss/base';
@import 'tailwindcss/components';
@import 'tailwindcss/utilities';
/**
 * END --- SETUP TAILWINDCSS EDIT
 */
body {
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
}
ul {
  list-style-type: none;
  margin: 1rem 0;
  padding: 0;
}
li {
  display: inline-block;
  margin: 0 1rem 0 0 ;
}
h1 > a {
  text-decoration: none;
  color: black;
}
button, input, label, textarea {
  display: block;
  outline: none;
}
label {
  margin-top: 1rem;
}
.error {
  color: red;
}
input.error, textarea.error {
  border: 1px solid red;
}
.form-error {
  color: red;
  background-color: lavenderblush;
  padding: 1rem;
  display: inline-block;
}
.form-error ul {
  list-style-type: disc;
  margin: 1rem;
  padding: 1rem;
}
.form-error li {
  display: list-item;
}
.flex-between {
  display: flex;
  justify-content: space-between;
}
.flex-between button {
  display: inline;
}
