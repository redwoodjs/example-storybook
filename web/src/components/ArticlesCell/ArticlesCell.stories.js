import { Loading, Empty, Failure, Success } from './ArticlesCell'
import { standard } from './ArticlesCell.mock'

export const loading = (args) => {
  return Loading ? <Loading {...args} /> : null
}

export const empty = (args) => {
  return Empty ? <Empty {...args} /> : null
}

export const failure = (args) => {
  return Failure ? <Failure error={new Error('Oh no')} {...args} /> : null
}

export const success = (args) => {
  return Success ? <Success {...standard()} {...args} /> : null
}

export default { title: 'Cells/ArticlesCell' }
