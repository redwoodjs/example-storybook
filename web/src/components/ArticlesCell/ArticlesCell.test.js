import { render } from '@redwoodjs/testing/web'
import { Loading, Empty, Failure, Success } from './ArticlesCell'
import { standard } from './ArticlesCell.mock'

// Generated boilerplate tests do not account for all circumstances
// and can fail without adjustments, e.g. Float and DateTime types.
//           Please refer to the RedwoodJS Testing Docs:
//        https://redwoodjs.com/docs/testing#testing-cells
// https://redwoodjs.com/docs/testing#jest-expect-type-considerations

describe('ArticlesCell', () => {
  it('renders Loading successfully', () => {
    expect(() => {
      render(<Loading />)
    }).not.toThrow()
  })

  it('Loading fragment matches snapshot', () => {
    const { asFragment } = render(<Loading />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders Empty successfully', async () => {
    expect(() => {
      render(<Empty />)
    }).not.toThrow()
  })

  it('Empty fragment matches snapshot', () => {
    const { asFragment } = render(<Empty />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders Failure successfully', async () => {
    expect(() => {
      render(<Failure error={new Error('Oh no')} />)
    }).not.toThrow()
  })

  it('Failure fragment matches snapshot', () => {
    const { asFragment } = render(<Failure error={new Error('Oh no')} />)
    expect(asFragment()).toMatchSnapshot()
  })

  // When you're ready to test the actual output of your component render
  // you could test that, for example, certain text is present:
  //
  // 1. import { screen } from '@redwoodjs/testing/web'
  // 2. Add test: expect(screen.getByText('Hello, world')).toBeInTheDocument()

  it('renders Success successfully', async () => {
    expect(() => {
      render(<Success articles={standard().articles} />)
    }).not.toThrow()
  })

  it('Success fragment matches snapshot', () => {
    const { asFragment } = render(<Success articles={standard().articles} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
