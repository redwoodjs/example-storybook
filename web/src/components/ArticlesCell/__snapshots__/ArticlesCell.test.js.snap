// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlesCell Empty fragment matches snapshot 1`] = `
<DocumentFragment>
  <div>
    Empty
  </div>
</DocumentFragment>
`;

exports[`ArticlesCell Failure fragment matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="color: red;"
  >
    Error: Oh no
  </div>
</DocumentFragment>
`;

exports[`ArticlesCell Loading fragment matches snapshot 1`] = `
<DocumentFragment>
  <div>
    Loading...
  </div>
</DocumentFragment>
`;

exports[`ArticlesCell Success fragment matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="space-y-10"
  >
    <article>
      <header>
        <h2
          class="text-xl text-blue-700 font-semibold"
        >
          <a
            href="/article/42"
          >
            ArticlesCell Title 1
          </a>
        </h2>
      </header>
      <div
        class="mt-2 text-gray-900 font-light"
      >
        ArticlesCell Body 1
      </div>
      <div>
        Posted at: 2020-01-01T00:00:00.000Z
      </div>
    </article>
    <article>
      <header>
        <h2
          class="text-xl text-blue-700 font-semibold"
        >
          <a
            href="/article/43"
          >
            ArticlesCell Title 2
          </a>
        </h2>
      </header>
      <div
        class="mt-2 text-gray-900 font-light"
      >
        ArticlesCell Body 2
      </div>
      <div>
        Posted at: 2020-01-02T00:00:00.000Z
      </div>
    </article>
    <article>
      <header>
        <h2
          class="text-xl text-blue-700 font-semibold"
        >
          <a
            href="/article/44"
          >
            ArticlesCell Title 3
          </a>
        </h2>
      </header>
      <div
        class="mt-2 text-gray-900 font-light"
      >
        ArticlesCell Body 3
      </div>
      <div>
        Posted at: 2020-01-03T00:00:00.000Z
      </div>
    </article>
  </div>
</DocumentFragment>
`;
